src/Compilers/CSharp/Test/Symbol/Compilation/GetUnusedImportDirectivesTests.cs

```diff
@@ -102,6 +102,72 @@ static void Main(string[] args)
             model.GetDiagnostics().Verify(Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using ClassLibrary2;"));
         }
 
+        [Fact]
+        public void TestUnusedExtensionMarksImportsAsUsed_02()
+        {
+            // new extension method
+            string class1Source = """
+using System;
+
+namespace ClassLibrary1
+{
+    public class Class1
+    {
+        public void Method1(string arg1)
+        {
+            Console.WriteLine(arg1);
+        }
+    }
+}
+""";
+            var classLib1 = CreateCompilation(source: class1Source, assemblyName: "ClassLibrary1");
+
+            string class2Source = """
+using System;
+using ClassLibrary1;
+
+namespace ClassLibrary2
+{
+    public static class Class2
+    {
+        extension(Class1 arg0)
+        {
+            public void Method1(string arg1)
+            {
+                Console.Write("Erroneous: " + arg1);
+            }
+        }
+    }
+}
+""";
+            var classLib2 = CreateCompilation(source: class2Source, assemblyName: "ClassLibrary2", references: [classLib1.ToMetadataReference()]);
+
+            string consoleApplicationSource = """
+using ClassLibrary2;
+using ClassLibrary1;
+
+var instance1 = new Class1();
+instance1.Method1("Argument1");
+""";
+            var tree = Parse(consoleApplicationSource);
+            var comp = CreateCompilation(tree, [classLib1.ToMetadataReference(), classLib2.ToMetadataReference()], assemblyName: "ConsoleApplication");
+            var model = comp.GetSemanticModel(tree) as CSharpSemanticModel;
+
+            var syntax = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single().Expression;
+
+            //This is the crux of the test.
+            //Without this line, with or without the fix, the model never gets pushed to evaluate extension method candidates
```

Comment:
>with or without the fix [](http://example.com/codeflow?start=33&length=23)

It is not clear what fix we are referring to. It looks like addition of this test corresponds to a removal of a comment in implementation.

Reply:
Replaced this with a more compact/focused test (`LookupSymbols_UsedImports`)